pod-operation/src/state_machine.rs

```diff
@@ -161,7 +167,16 @@ impl StateMachine {
 	/// Perform operations when the pod is running
 	fn _running_periodic(&mut self) -> State {
 		info!("Rolling Running state");
-		// TODO: add actual decision logic
+		let default_readings: [f32; 4] = self.ads1015_1.read_lim_temps().into();
+		let alternative_readings: [f32; 4] = self.ads1015_2.read_lim_temps().into();
+		let all_readings = [default_readings, alternative_readings].concat();
+		if all_readings
+			.iter()
+			.any(|&reading| reading > LIM_TEMP_THRESHOLD)
+		{
+			return State::Halted;
```

Comment:
Suggestion: Also as noted in #55, could we create a new `Faulted` state to indicate the pod decided to terminate itself due to a fault rather than being halted by operator intervention?

Reply:
yeah, would that have its own enter actions or would it end up being identical to halted?

Reply:
Will handle separately.